administrator/components/com_media/resources/scripts/components/modals/create-folder-modal.vue

```diff
@@ -23,13 +23,30 @@
     export default {
         name: 'create-folder-modal',
         directives: { focus: focus },
+        ready: function() {
+            /*add event listener on keydown*/
+            document.addEventListener("keydown", (e) => {
+                /*checks if the model is shown and enter(keycode 13) is pressed*/
+                if($store.state.showCreateFolderModal && e.keyCode == 13){
+                    save();
+                    close();
+                }
+            });
+        },
         methods: {
             /* Close the modal instance */
             close() {
                 this.$store.commit(types.HIDE_CREATE_FOLDER_MODAL);
             },
             /* Save the form and create the folder */
             save() {
+                /*Checks for the folder name, if it is not declared display an error to user and abort task*/
+                if(!this.folder){
+                    this.$store.commit(types.HIDE_CREATE_FOLDER_MODAL);
```

Comment:
Hm. Do we really want to close the modal when an validation error occurs? 

If the user submits the form, he most probably dont want the modal to be closed. He can press ESC key or close.

I believe you close the modal because the error message is behind the modal? Imo this is a css issue which should be fixed or solved in a different way. Closing the modal is not very user friendly.

Perhaps we can even use the 2 argument of Joomla.renderMessages (The selector of the container where the message will be rendered). 

Additionally we need to mark the input field as invalid (red border, label).